### PR TITLE
test suite

### PR DESCRIPTION
**Pull request by FredTheDino**
Add some more tests and update the bench-file.

Nothing much, but it should be enough to get information
about the performance of the interpeter.

### CODE REVIEW

**Comment by FredTheDino** [comment]
This is part of the way to the target, but it's not as much as I want it to be. I would love to see some more tests.

File: progs/bench/fib_iter.sy [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// A fibonacci implementation that is a little
+// less awefull. But we run it 1000 times instead.
```
Comment (sornas) [inline progs/bench/fib_iter.sy:2]:
```suggestion
// less awful. But we run it 1000 times instead.
```

File: progs/bench/fib.sy [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// The worst implementation of fibonacci calculations
```
Comment (sornas) [inline progs/bench/fib.sy:1]:
```suggestion
// The worst implementation of Fibonacci calculations
```

File: progs/bench/fib_iter.sy [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// A fibonacci implementation that is a little
```
Comment (sornas) [inline progs/bench/fib_iter.sy:1]:
```suggestion
// A Fibonacci implementation that is a little
```

File: src/vm.rs [hunk 4]
```diff
@@ -480,8 +480,9 @@ impl VM {
             self.frame().block.borrow().ops[self.frame().ip]);
     }
 
-    // Initalizes the VM for running. Run cannot be called before this.
-    pub(crate) fn init(&mut self, prog: &Prog) {
+    /// Initalizes the VM for running. Run cannot be called before this.
+    /// You do not have to call this.
+    pub fn init(&mut self, prog: &Prog) {
```
Comment (sornas) [inline src/vm.rs:485]: I feel like this should have a `#[doc(hidden)]` instead

**Review by sornas — APPROVED** [review]

**Pull request merged**